tests/e2e/e2e.go

```diff
@@ -65,57 +65,60 @@ func pingSvc(url string) {
 // waitForCmdOut runs a command until it gets
 // the expected output.
 // It accepts 2 arguments, cmd (command to be run)
-// & expOut (expected output).
+// timeout (the time to wait for the output)
+// check (function with output check logic)
 // It times out if the command doesn't fetch the
-// expected output  within the timeout period (1m).
-func waitForCmdOut(cmd string, expOut string) bool {
+// expected output  within the timeout period.
+func waitForCmdOut(cmd string, timeout int, check func(output string) bool) bool {
 
-	pingTimeout := time.After(1 * time.Minute)
+	pingTimeout := time.After(time.Duration(timeout) * time.Minute)
 	tick := time.Tick(time.Second)
 
 	for {
 		select {
 		case <-pingTimeout:
-			Fail("Timeout out after 1 minute")
+			Fail("Timeout out after " + string(timeout) + " minutes")
 
 		case <-tick:
 			out, err := exec.Command("/bin/sh", "-c", cmd).Output()
 			if err != nil {
 				Fail(err.Error())
 			}
 
-			if string(out) == expOut {
+			if check(string(out)) {
 				return true
 			}
 		}
 	}
+
 }
 
-// waitForDeleteCmd runs a command until it finds
-// the deleted resource to be absent in the list of relevant objects from cmd executed
-// It accepts 2 arguments, cmd (command to be run)
-// & object (the deleted object).
-// It times out if the command doesn't fetch the
-// expected output  within the timeout period (1m).
-func waitForDeleteCmd(cmd string, object string) bool {
+// waitForEqualCmd calls the waitForCmdOut function to wait and check if the output is equal to the given string within 1 min
+// cmd is the command to run
+// expOut is the expected output
+func waitForEqualCmd(cmd string, expOut string) bool {
```

Comment:
wouldn't timeout be required here as a parameter? why go with a default ;)
But would it be better to just implement checkCmdOutputEqual and may be checkCmdOutputContains and the caller just passes these standard func as args? only to avoid many wait funcs...
But it might be possible that, I am missing something

Reply:
1 minute timeout is a long time to wait according to me so it's better to have a predefined one for now. For services 5 minutes is more than enough, so kept it by default to 5 minutes. Otherwise we would have to pass these values again and again which would be harder to maintain. It would though us customise able  behaviour but for now only a few specific tests are using it

> But would it be better to just implement checkCmdOutputEqual and may be checkCmdOutputContains and the caller just passes these standard func as args? only to avoid many wait funcs

I kept it this way to have customise able behaviour in the future and also have developer friendly names in the main script.